goblet/resources/plugins/pydantic.py

```diff
@@ -0,0 +1,70 @@
+from typing import Any, Optional
+
+from apispec import BasePlugin, APISpec
+from pydantic import BaseModel
+
+
+class PydanticPlugin(BasePlugin):
+    def __init__(self) -> None:
+        super().__init__()
+        self.spec: APISpec | None = None
+
+    def init_spec(self, spec: APISpec) -> None:
+        super().init_spec(spec)
+        self.spec = spec
+
+    def schema_helper(
+        self,
+        name: str,
+        definition: dict,
+        model: Optional[BaseModel] = None,
+        **kwargs: Any
+    ):
+        if model is None:
+            return None
+
+        schema = model.schema(ref_template="#/components/schemas/{model}")
+
+        # if "definitions" in schema:
+        #     for (k, v) in schema["definitions"].items():
+        #         self.spec.components.schema(k, v)
+        #     del schema["definitions"]
```

Comment:
I don't know if this is necessary so I commented it out, but I also don't know how to test for this.

Reply:
yup it was needed! added that back in and the openapi spec was valid